Zend/tests/float_to_int/union_int_string_type_arg.phpt

```diff
@@ -0,0 +1,28 @@
+--TEST--
+Union of int|string shouldn't warn if string semantics are used
+--FILE--
+<?php
+
+$float = 1.0;
+
+function foo(int|string $a): void {
+    var_dump($a);
+}
+
+foo(1.0);
+foo(1.5);
+foo(fdiv(0, 0));
+foo(10e120);
+foo(10e500); // Infinity
+
+
+
+?>
+--EXPECTF--
+int(1)
+
+Deprecated: Implicit conversion from non-compatible float 1.5 to int in %s on line %d
```

Comment:
Interesting case. Would I be right in thinking that in PHP 9, the value is going to be (silently) treated as a string instead?